runner/engine_test.go

```diff
@@ -185,6 +185,28 @@ func TestRerunWhenFileChanged(t *testing.T) {
 	}
 }
 
+func TestRunCommand(t *testing.T) {
+	engine, err := NewEngine("", true)
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+	err = engine.runCommand("echo Hello Air")
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+}
+
+func TestRunPreCmd(t *testing.T) {
+	engine, err := NewEngine("", true)
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+	err = engine.runPreCmd()
```

Comment:
You should use a real command for integration testing, such as writing a file to the test folder and then checking if the file exists after execution is complete

Reply:
I don't think the current unit tests can test anything